python/cudf/cudf/core/_base_index.py

```diff
@@ -1348,6 +1348,15 @@ def isin(self, values):
         array([ True, False, False])
         """
 
+        # Even though only list-like objects are supposed to be passed, only
+        # scalars throw errors. Other types (like dicts) just transparently
+        # return False (see the implementation of ColumnBase.isin).
```

Comment:
what does pandas do here?

Reply:
This is describing the pandas behavior. For example:
```
>>> x = pd.Index([1, 2, 3])
>>> x.isin(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/vyasr/local/rapids/compose/etc/conda/cuda_11.5/envs/rapids/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 5632, in isin
    return algos.isin(self._values, values)
  File "/home/vyasr/local/rapids/compose/etc/conda/cuda_11.5/envs/rapids/lib/python3.8/site-packages/pandas/core/algorithms.py", line 459, in isin
    raise TypeError(
TypeError: only list-like objects are allowed to be passed to isin(), you passed a [int]
>>> x.isin([1])
array([ True, False, False])
>>> x.isin({'a': 1})
array([False, False, False])
```

Reply:
It might be worth mentioning that this behavior matches Pandas in the comment, otherwise someone might take it as advice on how to improve the code.

Reply:
Can do